### PR TITLE
Have script store return nullptr on facebook::jsi::JSINativeException

### PR DESCRIPTION
**Pull request by JunielKatarn**
Expected error conditions such as prepared script cache not (yet) existing were not handled, crashing the host process.

This change returns `nullptr` on memory-mapped buffer creation for subsequent handling by the JS engine.

Requesting backport to unblock Office feature.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8068)

### CODE REVIEW

File: vnext/Shared/BaseScriptStoreImpl.cpp [hunk 1]
```diff
@@ -147,7 +147,11 @@ std::unique_ptr<const jsi::Buffer> LocalFileSimpleBufferStore::getBuffer(const s
   }
 
   if (Microsoft::React::GetRuntimeOptionBool("JSI.MemoryMappedScriptStore")) {
-    return Microsoft::JSI::MakeMemoryMappedBuffer(winrt::to_hstring(storeDirectory_ + bufferId).c_str());
+    try {
+      return Microsoft::JSI::MakeMemoryMappedBuffer(winrt::to_hstring(storeDirectory_ + bufferId).c_str());
+    } catch (const facebook::jsi::JSINativeException &e) {
```
Comment (NikoAri) [inline vnext/Shared/BaseScriptStoreImpl.cpp:152]: >&e [](http://example.com/codeflow?start=53&length=2)

Should there be any telemetry about this error? Similar to ShipAssert?

**Review by NikoAri — APPROVED** [review]
:shipit:

**Review by hansenyy — APPROVED** [review]
:shipit:

File: vnext/Shared/BaseScriptStoreImpl.cpp (hunk 1 above)
Comment (JunielKatarn) [reply vnext/Shared/BaseScriptStoreImpl.cpp:152]: The non-memory-mapped approach doesn't include such telemetry, so don't know if this is the ideal spot for such counts/measurements.

Also, I don't think we can do shipasserts in RNW (non-Office) code.

**Pull request merged**